src/erlcloud_aws.erl

```diff
@@ -249,6 +250,111 @@ update_config(#aws_config{} = Config) ->
+    service_config( <<"cloudformation">>, Region, Config );
+service_config( <<"cloudtrail">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ cloudtrail_host = Host };
+service_config( <<"dynamodb">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ ddb_host = Host };
+service_config( <<"ddb">>, Region, Config ) ->
+    service_config( <<"dynamodb">>, Region, Config );
+service_config( <<"streams.dynamodb">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ ddb_streams_host = Host };
+service_config( <<"ec2">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ ec2_host = Host };
+service_config( <<"elasticloadbalancing">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ elb_host = Host };
+service_config( <<"elb">>, Region, Config ) ->
+    service_config( <<"elasticloadbalancing">>, Region, Config );
+service_config( <<"iam">>, _Region, Config ) -> Config;
+service_config( <<"kinesis">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ kinesis_host = Host };
+service_config( <<"mechanicalturk">>, _Region, Config ) -> Config;
+service_config( <<"rds">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ rds_host = Host };
+service_config( <<"s3">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ s3_host = Host };
+service_config( <<"sdb">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ sdb_host = Host };
+service_config( <<"ses">> = Service, Region, Config ) ->
+    Host = service_host( <<"email">>, Region ),
+    Config#aws_config{ ses_host = Host };
+service_config( <<"sns">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ sns_host = Host };
+service_config( <<"sqs">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ sqs_host = Host };
+service_config( <<"sts">> = Service, Region, Config ) ->
+    Host = service_host( Service, Region ),
+    Config#aws_config{ sts_host = Host };
+service_config( <<"waf">>, _Region, Config ) -> Config.
+
+    
+%%%---------------------------------------------------------------------------
+-spec service_host( Service :: atom() | binary(),
+                    Region :: binary() ) -> string().
+%%%---------------------------------------------------------------------------
+%% @hidden
+%% @doc Host name for the specified AWS service & region
+%%
+%% This function handles the special and general cases of service host
+%% names.
+%%
+service_host( <<"s3">>, <<"us-east-1">> ) -> "s3-external-1.amazonaws.com";
+service_host( <<"s3">>, Region ) ->
```

Comment:
@fp along with `us-east-1` there is one more exception to be added here for China region. 
https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_aws.erl#L112-L128.

as for gov cloud, I assume it will work with `<<"us-gov-west-1">> == Region` 
http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region